WalletWasabi/Crypto/GroupElementVector.cs

```diff
@@ -30,5 +30,34 @@ internal GroupElementVector(params GroupElement[] groupElements)
 			GetEnumerator();
 
 		public int Count => GroupElements.Count();
+
+		public static bool operator ==(GroupElementVector? x, GroupElementVector? y) => x?.Equals(y) ?? false;
```

Comment:
@nothingmuch I have added in commit f765919d1f2e91e291ed86bd114d2ecb2c3fb9a5 ("Implement equality rules.") equality rules for various classes. Now the first question is whether it is a good idea or not. I honestly don't know. 

I would be much happier if implementing of `operator ==` were not necessary but `MemoryCache` implementation forces me to do it due to its internal implementation. Otherwise, the comparison of objects fail.

Obviously, the approach taken may be wrong and we may try something else.

Anyway, any feedback on this would help me to move this PR forward as currently I'm stuck on this decision.

Reply:
Based on discussion with Yuval IRL, he agreed the idea is good.